src/plugin/settings/engine.cpp

```diff
@@ -299,37 +309,45 @@ Result<const ISettingsResponse *> Engine::settingsReceived()
         {
             if (!fs::create_directories(fileCacheDir, errCode))
             {
+                NX_PRINT <<  "Unable to create file cache directory " + fileCacheDir + " with error: " + errCode.message() << std::endl;
                 return error(ErrorCode::internalError, "Unable to create file cache directory " + fileCacheDir +
                                                            " with error: " + errCode.message());
             }
             fs::permissions(fileCacheDir, fs::perms::all, fs::perm_options::add, errCode);
             if (errCode)
             {
+                NX_PRINT <<  "Unable to set file cache directory permissions with error: " + errCode.message() << std::endl;
                 return error(ErrorCode::internalError,
                              "Unable to set file cache directory permissions with error: " + errCode.message());
             }
         }
 
         if (!cfManager.isInstalled())
         {
+            NX_PRINT <<  "Cloudfuse is not installed" << std::endl;
             return error(ErrorCode::internalError, "Cloudfuse is not installed");
         }
 
 #if defined(__linux__)
+        NX_PRINT <<  "spawning process from genS3Config" << std::endl;
```

Comment:
You could pull this print out of the `#if defined` block so you don't have to repeat it.
Same for the other prints before platform-specific calls.